src/Data/List/Relation/Suffix/Heterogeneous/Properties.agda

```diff
@@ -0,0 +1,233 @@
+              Pointwise R (List.foldl (flip _∷_) as′ as) (List.foldl (flip _∷_) bs′ bs)
+      foldl rs′ []       = rs′
+      foldl rs′ (r ∷ rs) = foldl (r ∷ rs′) rs
+
+  fromPrefix⁺ : ∀ {as bs} → Prefix R as bs → Suffix R (reverse as) (reverse bs)
+  fromPrefix⁺ {as} {bs} p with Prefix.toView p
+  ... | Prefix._++_ {cs} rs ds = subst (Suffix R (reverse as))
+                                       (sym (Listₚ.reverse-++-commute cs ds))
+                               $ Suffix.fromView (reverse ds Suffix.++ pw-reverse rs)
+
+  fromPrefix⁻ : ∀ {as bs} → Prefix R (reverse as) (reverse bs) → Suffix R as bs
+  fromPrefix⁻ pre = P.subst₂ (Suffix R) (Listₚ.reverse-involutive _) (Listₚ.reverse-involutive _)
+                  $ fromPrefix⁺ pre
+
+  toPrefix⁺ : ∀ {as bs} → Suffix R as bs → Prefix R (reverse as) (reverse bs)
+  toPrefix⁺ {as} {bs} s with Suffix.toView s
+  ... | Suffix._++_ cs {ds} rs = subst (Prefix R (reverse as))
+                                       (sym (Listₚ.reverse-++-commute cs ds))
+                               $ Prefix.fromView (pw-reverse rs Prefix.++ reverse cs)
+
+  toPrefix⁻ : ∀ {as bs} → Suffix R (reverse as) (reverse bs) → Prefix R as bs
+  toPrefix⁻ suf = P.subst₂ (Prefix R) (Listₚ.reverse-involutive _) (Listₚ.reverse-involutive _)
+                $ toPrefix⁺ suf
+
+------------------------------------------------------------------------
+-- length
+
+module _ {a b r} {A : Set a} {B : Set b} {R : REL A B r} {as} where
+
+  length-mono-Suffix-≤ : ∀ {bs} → Suffix R as bs → length as ≤ length bs
+  length-mono-Suffix-≤ (here rs)   = ℕₚ.≤-reflexive (Pointwise-length rs)
+  length-mono-Suffix-≤ (there suf) = ℕₚ.≤-step (length-mono-Suffix-≤ suf)
+
+------------------------------------------------------------------------
+-- Pointwise conversion
+
+module _ {a b r} {A : Set a} {B : Set b} {R : REL A B r} where
+
+  fromPointwise : Pointwise R ⇒ Suffix R
+  fromPointwise = here
+
+  toPointwise : ∀ {as bs} → length as ≡ length bs → Suffix R as bs → Pointwise R as bs
+  toPointwise eq (here rs) = rs
+  toPointwise eq (there suf) =
+    let as≤bs = length-mono-Suffix-≤ suf
+        as>bs = ℕₚ.≤-reflexive (sym eq)
+    in contradiction as≤bs (ℕₚ.<⇒≱ as>bs)
+
+------------------------------------------------------------------------
+-- Suffix as a partial order
+
+module _ {a b c r s t} {A : Set a} {B : Set b} {C : Set c}
+         {R : REL A B r} {S : REL B C s} {T : REL A C t} where
+
+  trans : Trans R S T → Trans (Suffix R) (Suffix S) (Suffix T)
+  trans rs⇒t (here rs)    (here ss)       = here (Pw.transitive rs⇒t rs ss)
+  trans rs⇒t (here rs)    (there ssuf)    = there (trans rs⇒t (here rs) ssuf)
+  trans rs⇒t (there rsuf) ssuf            = trans rs⇒t rsuf (tail ssuf)
+
+module _ {a b e r s} {A : Set a} {B : Set b}
+         {R : REL A B r} {S : REL B A s} {E : REL A B e} where
+
+  private
+    pw-antisym : Antisym R S E → Antisym (Pointwise R) (Pointwise S) (Pointwise E)
```

Comment:
This should also go in `Pointwise`. May be worth creating `Pointwise.Properties` now.

Reply:
Same for `pw-map⁺`, etc.